### PR TITLE
Named broker

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by tribal-tec** [comment]
More descriptive commit message please

File: zeq/connection/broker.h [hunk 1]
```diff
@@ -24,6 +35,35 @@ namespace detail { class Broker; }
 class Broker : public Receiver
 {
 public:
+    enum AllocationMode
```
Comment (tribal-tec) [inline zeq/connection/broker.h]: more PortSelection or why allocation?

File: zeq/connection/service.cpp [hunk 2]
```diff
@@ -63,5 +64,17 @@ bool Service::subscribe( const std::string& brokerAddress,
     return address == std::string( result );
 }
 
+bool Service::subscribe( const std::string& brokerAddress,
+                         const std::string& name,
+                         const Publisher& publisher )
+{
+    const lunchbox::uint128_t& md5 = lunchbox::make_uint128( name );
+    const uint16_t port = 1024 + (md5.low() % (65535-1024));
+    std::string address( brokerAddress + ":" +
```
Comment (tribal-tec) [inline zeq/connection/service.cpp]: const?

Comment (tribal-tec) [inline zeq/connection/service.cpp]: 71-73 are C&P from broker ctor. please refactor

File: zeq/connection/broker.h (hunk 1 above)
Comment (eile) [inline zeq/connection/broker.h]: Was PortAllocation before. Will find a better name.

File: zeq/connection/service.cpp (hunk 2 above)
Comment (eile) [inline zeq/connection/service.cpp]: Actually only 71-72. Was reluctant to create a function for that.

**Comment by eile** [comment]
All done.

**Comment by tribal-tec** [comment]
retest this please

**Comment by eile** [comment]
Can't login to Jenkins... Can you C&P the failing test output?

**Comment by tribal-tec** [comment]
```
Stacktrace

Running 4 test cases...
  PID.Thread   | Filename:line                  | ms | Message
/gpfs/bbp.cscs.ch/home/bbprelman/jenkins-cscs-01/workspace/oss.zeq.github/build_type/Release/platform/cscsviz/tests/connection/broker.cpp(138): error in "test_named_broker_port_used": exception std::runtime_error is expected

*** 1 failure detected in test suite "zeq_connection_broker"
Standard Output

Running 4 test cases...
  PID.Thread   | Filename:line                  | ms | Message
/gpfs/bbp.cscs.ch/home/bbprelman/jenkins-cscs-01/workspace/oss.zeq.github/build_type/Release/platform/cscsviz/tests/connection/broker.cpp(138): error in "test_named_broker_port_used": exception std::runtime_error is expected

*** 1 failure detected in test suite "zeq_connection_broker"
```

**Comment by eile** [comment]
retest this please

**Comment by eile** [comment]
@jafyvilla Please review

**Comment by jafyvilla** [comment]
+1. There seems to be a problem with Jenkins/github builds though

**Comment by eile** [comment]
retest this please

**Pull request merged**